.github/workflows/python-package.yml

```diff
@@ -0,0 +1,46 @@
+# This workflow will install Python dependencies, run tests and lint with a variety of Python versions
+# For more information see: https://docs.github.com/en/actions/automating-builds-and-tests/building-and-testing-python
+
+name: Python package
+
+on:
+  push:
+    branches: [ "main" ]
+  pull_request:
+    branches: [ "main" ]
+
+jobs:
+  build:
+
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+      matrix:
+        python-version: ["3.9", "3.10", "3.11"]
+        poetry-version: [2.1.1]
```

Comment:
FYI, dowhy is using an older poetry version, 1.5.1. I assume there's no potential conflict since it's not a runtime dependency.

Reply:
yes, that should not be a conflict since this is used only for dev/testing.